pkgs/top-level/all-packages.nix

```diff
@@ -17793,7 +17793,9 @@ with pkgs;
   coq_8_4 = coqPackages_8_4.coq;
   coq_8_5 = coqPackages_8_5.coq;
   coq_8_6 = coqPackages_8_6.coq;
-  coq_HEAD = callPackage ../applications/science/logic/coq/HEAD.nix {};
+  coq_HEAD = callPackage ../applications/science/logic/coq {
+    version = "8.7pre-2017-05-28";
```

Comment:
For the case of post June 15th / June 1st, with the `8.7beta1` / `8.7pre` being added, yes, there will be `coq_8_7_beta`, but a new snapshot several commits after this tag still resides in `coq_HEAD` (which is the point of this PR).
For the in-between case, it is not enough to declare a version into a snapshot of a repo when the version hasn't been described in a git tag, only written in a roadmap / spoken unofficially.